### ISSUE TITLE
Reintroduce web2.dav in twisted.web

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3081]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)| @wsanchez reported|
|-|-|
|Trac ID|trac#3081|
|Type|enhancement|
|Created|2008-03-16 17:10:56Z|
|Branch|https://github.com/twisted/twisted/tree/dav-take-two-3081-5|

Reintroduce web2.dav in trunk.  This is gated on web2 becoming stable.

<details><summary>Searchable metadata</summary>

```
trac-id__3081 3081
type__enhancement enhancement
reporter__wsanchez wsanchez
priority__normal normal
milestone__ 
branch__branches_dav_take_two_3081_5 branches/dav-take-two-3081-5
branch_author__exarkun__dreid__glyph exarkun, dreid, glyph
status__closed closed
resolution__wontfix wontfix
component__web web
keywords__ 
time__1205687456000000 1205687456000000
changetime__1560027754338322 1560027754338322
version__None None
owner__ 
cc__dreid cc__glyph cc__exarkun cc__rainhead cc__chris-
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_1"></a>|@wsanchez commented|
|-|-|

(In [22844]) Branch for DAV ongoing work to be brought back to trunk after web2 is stable.

Refs [#3081](https://github.com/twisted/twisted/issues/3081)

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_3"></a>|@dreid commented|
|-|-|

(In [23609]) Branching to 'dav-take-two-3081-2'

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_4"></a>|@dreid commented|
|-|-|

(In [23620]) Branching to 'dav-take-two-3081-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [26342]) Branching to 'dav-take-two-3081-3'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [26739]) Branching to 'dav-take-two-3081-4'

**Comment by twisted-trac** [comment]
|<img alt="davidsarah's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|davidsarah commented|
|-|-|

What's the progress on this? I would really like to use a WebDAV implementation that is integrated with twisted (for Tahoe-LAFS), but the [TwistedWeb2](http://twistedmatrix.com/trac/wiki/TwistedWeb2) page seems to suggest that "web2 becoming stable" isn't going to happen.

Would using [WsgiDAV](http://code.google.com/p/wsgidav/) on top of twisted.web.wsgi be a reasonable or recommended thing to do at this point?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

It's not clear what the plan is here.  "web2 becoming stable" definitely isn't condition that makes sense anymore.  Ultimately, the plan should be to implement something on top of Twisted Web.  wsanchez or glyph are more likely to have some idea of how to accomplish this.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_9"></a>|@wsanchez set owner to @glyph<br>@wsanchez set status to `new`|
|-|-|

The current plan is basically a rewrite on top of web. Glyph is working on the steps needed to get us there.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_10"></a>|@wsanchez commented|
|-|-|

I think that by the end of PyCon (this month) we should have a much clearer picture on where we stand on this.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_11"></a>|@glyph commented|
|-|-|

(In [28162]) Branching to 'dav-take-two-3081-5'

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_12"></a>|@glyph set status to `assigned`|
|-|-|

The current status of this work is that the DAV stuff is being reworked on calendarserver.org.  When it makes its re-debut in the Twisted tree, it will be a subpackage of web, not web2.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_17"></a>|@wsanchez set status to `closed`|
|-|-|

This is somewhat unlikely to happen.
